### PR TITLE
workflows: disable rollback on CLI install

### PR DESCRIPTION
**Pull request by nbusseneau**
By default, `cilium install` tries to rollback the installation if something goes wrong. We explicitly disable this behaviour in CI as this forbids us to retrieve Cilium's state if something goes wrong at install time.

### CODE REVIEW

**Review by joestringer — APPROVED** [review]
Thanks!

File: .github/workflows/conformance-aks.yaml [hunk 1]
```diff
@@ -20,7 +20,7 @@ on:
   #   will disappear from the PR checks: please provide a direct link to the
   #   successful workflow run (can be found from Actions tab) in a comment.
   # 
-  # pull_request: {}
+  pull_request: {}
```
Comment (joestringer) [inline .github/workflows/conformance-aks.yaml]: Ah right I guess we should wait for the results of this and revert this change before merging but otherwise LGTM :+1:

**Review by joestringer — CHANGES_REQUESTED** [review]

File: .github/workflows/conformance-aks.yaml (hunk 1 above)
Comment (nbusseneau) [reply .github/workflows/conformance-aks.yaml]: Yep. I actually intended to run the workflow a few times to see if I can reproduce the install issue over at #18107 and get logs.

Comment (pchaigno) [reply .github/workflows/conformance-aks.yaml]: IMO, we should put those workflow PR in draft until we're finished with testing.

**Comment by nbusseneau** [comment]
Link to successful run of workflow changes: https://github.com/cilium/cilium/actions/runs/1546043233/attempts/3
Link to failing run where we see the "no rollback" behaviour in action: https://github.com/cilium/cilium/actions/runs/1546043233/attempts/2

CI passed, removing temporary commit.

**Review by pchaigno — APPROVED** [review]

**Review by joestringer — APPROVED** [review]
